nservicebus/architecture/publishing-from-web-applications.md

```diff
@@ -0,0 +1,89 @@
+* Cloud transports generally have an associated cost per interaction. This means that requiring an unnecessary send before publish costs extra money to operate.
+* In NServiceBus 4.0 and 5.0, the process of creating a send-only endpoint (which is still capable of publishing messages, but does not have an input queue or process incoming messages) was made much easier.
+
+## Publish/Subscribe Mechanics
+
+First it's important to understand how publish/subscribe actually works, consisting of the subscription phase, and then the actual publish operation.
+
+### Subscription
+
+An endpoint must register its interest in a message by subscribing to a specific message type. How this occurs is dependent upon the message transport being used.
+
+Some transports (Azure Service Bus, RabbitMQ) support publish/subscribe natively. This means that when an endpoint wants to subcribe to an event, it contacts the broker directly, and the broker maintains the list of subscriptions.
+
+Other transports (MSMQ, SQL, Azure Storage Queues) do not support native publish/subscribe, and instead use storage-driven publishing with message-driven subscriptions. This means that each endpoint is responsible for maintaining its own subscription storage, usually in a database. When an endpoint wants to subscribe to an event, it sends a subscription request message to the owner endpoint, which will update its own subscription storage.
+
+### Publishing
+
+The publishing process also differs based on the message transport.
+
+A transport that supports native publish/subscribe (Azure Service Bus, RabbitMQ) will send the event message directly to the broker, addressed to a topic, which will distribute the message to all interested parties.
+
+A transport that uses storage-driven publishing will act differently. When it needs to publish, it will first consult its subscription storage to determine the interested parties. It will then send an independent copy of the event message to each subscriber.
+
+While it is common to think of an event being published *from* a specific location, this doesn't always reflect the reality of the messaging mechanics. For native pub/sub brokers, the "publish from" location (the `Endpoint` value in the `MessageEndpointMappings` section) relates to a Topic in the broker.
+
+In the case of transports, however, the "publish from" location actually relates to the input queue that is set up to receive subscription requests.
+
+## Load Balancing
+
+Typically, web applications will be scaled out with a network load balancer either to handle large amounts of traffic, or for high availability, or both. This is further complicated by elastic cloud scenarios, where additional instances of a web role can be provisioned and later removed, either manually or automatically, to keep up with a changing amount of load.

```

Comment:
I think you're on the wrong paragraph? This is still important because it sets up the conditions a scaled out web app has to satisfy to all publish as one logical publisher.

Right?